Calibration/HcalAlCaRecoProducers/plugins/AlCaIsoTracksFilter.cc

```diff
@@ -351,11 +359,17 @@ bool AlCaIsoTracksFilter::filter(edm::Event& iEvent, edm::EventSetup const& iSet
 				   << " Chg Isolation " << hmaxNearP
 				   << ":" << eIsolation;
 	  if (t_p>pTrackMin_ && eMipDR<eEcalMax_ && hmaxNearP<eIsolation) {
-	    accept = true;
-	    break;
+	    if (t_p > pTrackLow_ && t_p < pTrackHigh_) ntrin++;
+	    else                                       ntrout++;
 	  }
 	}
       }
+      accept = (ntrout > 0);
+      if (!accept && ntrin > 0) {
+	++nRange_;
+	if      (preScale_ <= 1)         accept = true;
+	else if (nRange_%preScale_ == 1) accept = true;
+      }
     }
   }
   // Step 4:  Return the acceptance flag
```

Comment:
hello @bsunanda 

it would help a bit if in the description of the PR you wrote the logic you're trying to implement

My understanding of this piece of code:

   * you count how many tracks match your Pt range
   * and how many tracks are	outside the Pt range you want

IF there are tracks **OUTSIDE your range** 
   * --> all objects get added to the persistable collections

OTHERWISE if there are tracks inside your range
   * --> you	add one object out of every preScale_   <==== is this what you want ?